superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DatasetSelect.tsx

```diff
@@ -68,9 +68,11 @@ const DatasetSelect = ({ onChange, value }: DatasetSelectProps) => {
         const list: {
           label: string | ReactNode;
           value: string | number;
+          customLabel: string;
         }[] = response.json.result.map((item: Dataset) => ({
           label: DatasetSelectLabel(item),
           value: item.id,
+          customLabel: item.table_name,
```

Comment:
@msyavuz @rebenitez1802 Previously, you could load the Select options with [label and customLabel](https://github.com/apache/superset/blob/49df35d01298c997b79a3369d754815dc25a2742/superset-frontend/src/components/Select/utils.tsx#L210). `label` was used to searching and `customLabel` to rendering the option using a React component. There were multiple places using the `customLabel` logic. We need to check if the Ant Design migration broke this logic.